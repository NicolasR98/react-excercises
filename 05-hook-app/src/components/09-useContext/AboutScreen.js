import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutScreen = () => {
	const { user, setUser } = useContext(UserContext);

	const handleLogout = () => {
		setUser({});
		console.log('user logged out');
	};

	return (
		<div>
			<h1>About Screen</h1>
			<hr />

			<pre className="container">{JSON.stringify(user, null, 3)}</pre>

			<button className="btn btn-warning" onClick={handleLogout}>
				Logout
			</button>
		</div>
	);
};
