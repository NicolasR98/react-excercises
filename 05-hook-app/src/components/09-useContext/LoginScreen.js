import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
	const { setUser } = useContext(UserContext);

	const testUser = {
		id: 1234,
		name: 'Nicolas',
	};

	const handleUserLogin = () => {
		setUser(testUser);
        console.log('user set');
	};

	return (
		<div>
			<h1>Login Screen</h1>
			<hr />
			<button className="btn btn-primary" onClick={handleUserLogin}>Login</button>
		</div>
	);
};
