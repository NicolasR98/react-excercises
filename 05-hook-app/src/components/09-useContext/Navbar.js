import React from 'react';
import { NavLink } from 'react-router-dom';

export const Navbar = () => {
	return (
		<nav className="navbar navbar-expand-sm navbar-dark bg-dark">
			<div className="container-fluid">
				<NavLink className="navbar-brand" to={'/'}>
					useContext
				</NavLink>
				<div className="collapse navbar-collapse" id="navbarNavAltMarkup">
					<div className="navbar-nav">
						<NavLink
                            exact={'true'}
							to={'/'}
							className={({ isActive }) =>
								'nav-link ' + (isActive ? 'active' : '')
							}
						>
							Home
						</NavLink>
						<NavLink
							className={({ isActive }) =>
								'nav-link ' + (isActive ? 'active' : '')
							}
							exact="true"
							to={'/login'}
						>
							Login
						</NavLink>
						<NavLink
							className={({ isActive }) =>
								'nav-link ' + (isActive ? 'active' : '')
							}
							exact="true"
							to={'/about'}
						>
							About
						</NavLink>
					</div>
				</div>
			</div>
		</nav>
	);
};
