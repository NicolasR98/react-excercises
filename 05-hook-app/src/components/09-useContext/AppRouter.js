import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { Navbar } from './Navbar';

export const AppRouter = () => {
	return (
		<Router>
			<div>
				<Navbar />
				<Routes>
					<Route exact path="/login" element={<LoginScreen />} />
					<Route exact path="/about" element={<AboutScreen />} />
					<Route exact path="/" element={<HomeScreen />} />
					<Route path="*" element={<HomeScreen />} />
				</Routes>
			</div>
		</Router>
	);
};
