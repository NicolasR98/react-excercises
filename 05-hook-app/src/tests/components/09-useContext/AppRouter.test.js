import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { AppRouter } from '../../../components/09-useContext/AppRouter';

describe('Tests on <AppRouter />', () => {
    const user = {
        id: 1,
        name: 'Nicolas',
    };

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
