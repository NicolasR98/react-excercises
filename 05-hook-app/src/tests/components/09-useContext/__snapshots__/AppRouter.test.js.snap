// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> should display correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            exact="true"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link "
            exact="true"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            className="nav-link "
            exact="true"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      Home Screen
    </h1>
    <hr />
    <pre
      className="container "
    >
      {
   "id": 1,
   "name": "Nicolas"
}
    </pre>
  </div>
</div>
`;
